stakerservice/service.go

```diff
@@ -561,3 +565,19 @@ func (s *StakerService) RunUntilShutdown(ctx context.Context) error {
 
 	return nil
 }
+
+// BasicAuthMiddleware handles the authentication of username and password
+// of this router
+func BasicAuthMiddleware(expUsername, expPwd string) func(http.HandlerFunc) http.HandlerFunc {
+	return func(next http.HandlerFunc) http.HandlerFunc {
+		return func(w http.ResponseWriter, r *http.Request) {
+			user, pass, ok := r.BasicAuth()
```

Comment:
I wonder whether we can take some insipiration in how auth is done from btcd - https://github.com/btcsuite/btcd/blob/cba88226f49d5162c83c3edb562cc926edb4a87f/rpcserver.go#L4150 🤔

Reply:
This seems more complex, but I can those maps to the staker service and have the `BasicAuthMiddleware` as a method to check basic auth

```go
	authsha                [sha256.Size]byte
	limitauthsha           [sha256.Size]byte
```